akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/scaladsl/TestEntityRefSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.scaladsl
+
+import scala.concurrent.Future
+
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import akka.Done
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.ActorRef
+import akka.cluster.ClusterSettings.DataCenter
+import akka.cluster.sharding.ShardCoordinator
+import akka.cluster.sharding.ShardCoordinator.LeastShardAllocationStrategy
+import akka.cluster.sharding.typed.ClusterShardingQuery
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.javadsl
+
+object TestEntityRefSpec {
+
+  final case class AskReq(s: String, replyTo: ActorRef[Done])
+
+  private val entityTypeKey1 = EntityTypeKey[String]("Key1")
+  private val entityTypeKey2 = EntityTypeKey[AskReq]("Key2")
+
+}
+
+class TestEntityRefSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike with LogCapturing {
+  import TestEntityRefSpec._
+
+  "TestEntityRef" must {
+    "send message with tell" in {
+      val probe = createTestProbe[String]()
+      val entityRef = TestEntityRef(entityTypeKey1, "entity-1", probe.ref)
+      entityRef ! "msg1"
+      entityRef ! "msg2"
+      probe.expectMessage("msg1")
+      probe.expectMessage("msg2")
+    }
+
+    "send message with ask" in {
+      val probe = createTestProbe[AskReq]()
+      val entityRef = TestEntityRef(entityTypeKey2, "entity-2", probe.ref)
+      val reply: Future[Done] = entityRef.ask[Done](replyTo => AskReq("req1", replyTo))
+      probe.receiveMessage().replyTo ! Done
+      reply.futureValue should ===(Done)
+    }
+
+    "be useful will ClusterShardingStub" in {
+      val probes = Map("entity-1" -> createTestProbe[String](), "entity-2" -> createTestProbe[String]())
+
+      class ClusterShardingStub extends javadsl.ClusterSharding with ClusterSharding {
```

Comment:
maybe we should make it easier to create a stub of `ClusterSharding` also?

Reply:
I haven't yet had the need for that, but this might become useful in the future.

Reply:
ok, let's add that later if/when needed